include/boost/multiprecision/cpp_int/checked.hpp

```diff
@@ -15,6 +21,28 @@ namespace boost { namespace multiprecision { namespace backends { namespace deta
 // Note that this is not a complete header, it must be included as part of boost/multiprecision/cpp_int.hpp.
 //
 
+template <typename T>
+inline constexpr T type_max(T) noexcept
```

Comment:
Stupid point of pedantry: the function parameter could be removed here and we could just call `type_max<T>()` instead.